internal/tasks/transform.go

```
@@ -0,0 +1,276 @@
+	}
+
+	// Return the transformed struct
+	return gen.ListOverdueTasksParams{
+		ListID: dbListID,
+		UserID: dbUserID,
+	}, nil
+}
+
+// toDBListTasksByStatusParams converts CountTasksByStatusParams (Go struct) into a pgtype-compatible struct for ListTasksByStatus.
+func toDBListTasksByStatusParams(params CountTasksByStatusParams) (gen.ListTasksByStatusParams, error) {
+	// Convert and validate ListID
+	dbListID, err := common.ToPgUUID(params.ListID)
+	if err != nil {
+		return gen.ListTasksByStatusParams{}, fmt.Errorf("invalid list_id: %w", err)
+	}
+
+	// Convert and validate UserID
+	dbUserID, err := common.ToPgUUID(params.UserID)
+	if err != nil {
+		return gen.ListTasksByStatusParams{}, fmt.Errorf("invalid user_id: %w", err)
+	}
+
+	// Convert Status to pgtype.Text
+	dbStatus := common.ToPgText(params.Status)
+
+	// Return the transformed struct for ListTasksByStatus query
+	return gen.ListTasksByStatusParams{
+		ListID: dbListID,
+		UserID: dbUserID,
+		Status: dbStatus,
+	}, nil
+}
+
+// toDBSearchTasksParams transforms SearchTasksParams into a database-compatible struct.
+func toDBSearchTasksParams(params SearchTasksParams) (gen.SearchTasksParams, error) {
+	// Convert ListID and UserID
+	dbListID, err := common.ToPgUUID(params.ListID)
+	if err != nil {
+		return gen.SearchTasksParams{}, fmt.Errorf("invalid list_id: %w", err)
+	}
+
+	dbUserID, err := common.ToPgUUID(params.UserID)
+	if err != nil {
+		return gen.SearchTasksParams{}, fmt.Errorf("invalid user_id: %w", err)
+	}
+
+	// Use common helper to safely convert Keyword to Text
+	dbKeyword := common.ToPgText(params.Keyword)
+
+	// Return the transformed struct
+	return gen.SearchTasksParams{
+		ListID:  dbListID,
+		UserID:  dbUserID,
+		Column3: dbKeyword,
+	}, nil
+}
+
+func toDBUpdatePriorityParams(params UpdateTaskParams) gen.UpdateTaskPriorityParams {
+	// Convert and validate TaskID
+	dbTaskID, _ := common.ToPgUUID(params.ID)
```

Should these errors be ignored no matter how unlikely? If you use an "_"  for ignoring an error you need to have a comment as to why you are ignoring it. It's OK to have this transform return an error.